src/Rector/v12/v3/flexform/MigrateItemsToIndexedArrayKeysForFlexFormItemsRector.php

```diff
@@ -0,0 +1,202 @@
+                $label = $numIndexes->item(0);
+                $value = $numIndexes->item(1);
+                $icon = $numIndexes->item(2);
+                $group = $numIndexes->item(3);
+                $description = $numIndexes->item(4);
+                if (! $label instanceof DOMElement) {
+                    continue;
+                }
+
+                $hasChanged = true;
+                $this->changeTagName($label, 'label');
+
+                if ($type->textContent !== 'check') {
+                    if (! $value instanceof DOMElement) {
+                        continue;
+                    }
+
+                    $hasChanged = true;
+                    $this->changeTagName($value, 'value');
+                }
+
+                if ($type->textContent === 'select') {
+                    if (! $icon instanceof DOMElement) {
+                        continue;
+                    }
+
+                    $hasChanged = true;
+                    $this->changeTagName($icon, 'icon');
+
+                    if (! $group instanceof DOMElement) {
+                        continue;
+                    }
+
+                    $hasChanged = true;
+                    $this->changeTagName($group, 'group');
+
+                    if (! $description instanceof DOMElement) {
+                        continue;
+                    }
+
+                    $hasChanged = true;
+                    $this->changeTagName($description, 'description');
+                }
+            }
+        }
+
+        return $hasChanged;
+    }
+
+    /**
+     * @codeCoverageIgnore
+     */
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Migrates indexed item array keys to associative for type select, radio and check in FlexForms. This Rector Rule is sponsored by UDG Rhein-Main GmbH',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+<select_single_1>
+    <label>select_single_1 description</label>
+    <description>field description</description>
+    <config>
+        <type>select</type>
+        <renderType>selectSingle</renderType>
+        <items>
+            <numIndex index="0">
+                <numIndex index="0">foo1</numIndex>
+                <numIndex index="1">foo1</numIndex>
+            </numIndex>
+            <numIndex index="1">
+                <numIndex index="0">foo2</numIndex>
+                <numIndex index="1">foo2</numIndex>
+            </numIndex>
+        </items>
+    </config>
+</select_single_1>
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+<select_single_1>
+    <label>select_single_1 description</label>
+    <description>field description</description>
+    <config>
+        <type>select</type>
+        <renderType>selectSingle</renderType>
+        <items>
+            <numIndex index="0">
+                <label>foo1</label>
```

Comment:
Is the example correct, label and value are the same?

Reply:
I copied it from here but can adjust it as well. 
https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.3/Deprecation-99739-IndexedArrayKeysForTCAItems.html